store/sqlstore/thread_store.go

```diff
@@ -133,17 +137,57 @@ func (s *SqlThreadStore) GetThreadsForUser(userId string, opts model.GetUserThre
 	if opts.PageSize == 0 {
 		pageSize = opts.PageSize
 	}
+
+	mentionsQuery, mentionsQueryArgs, _ := s.getQueryBuilder().
+		Select("SUM(ThreadMemberships.UnreadMentions)").
+		From("ThreadMemberships").
+		Join("Posts ON Posts.RootId = ThreadMemberships.PostId").
+		Join("Channels ON Posts.ChannelId = Channels.Id").
```

Comment:
You can join here with Threads tables and use the Join there to optain the ChannelId to join the Channels table., the table is going to be way less active and crowded than the Posts table.